src/UI/MainWindow.vala

```diff
@@ -109,6 +111,45 @@ public class Spreadsheet.UI.MainWindow : ApplicationWindow {
         show_all ();
     }
 
+    public MainWindow.with_state (Gtk.Application app, int x, int y, int w, int h, bool m) {
+        Object (application: app);
+        move (x, y);
```

Comment:
Maybe you could just call `move` and `maximize` (if needed) here, and move the common part of the two constructors in a `construct` block or in an `init` method that would be called by both constructors.